resources/lib/videoplayer.py

```diff
@@ -207,6 +212,59 @@ def _handle_skipping(self):
         utils.crunchy_log("_handle_skipping: starting thread", xbmc.LOGINFO)
         threading.Thread(target=self.thread_check_skipping).start()
 
+    def _handle_upnext(self):
+        try:
+            next_episode = self._stream_data.next_playable_item
+            if not next_episode:
+                utils.crunchy_log("_handle_upnext: No episode or disabled upnext integration")
+                return
+            next_url = view.build_url(
+                {
+                    "series_id": G.args.get_arg("series_id"),
+                    "episode_id": next_episode.episode_id,
+                    "stream_id": next_episode.stream_id
+                },
+                "video_episode_play"
+            )
+            show_next_at_seconds = self._compute_when_episode_ends()
+            # Needs to wait 1s, otherwise, upnext will show next dialog at episode start...
+            xbmc.sleep(1000)
+            utils.crunchy_log("_handle_upnext: Next URL (shown at %ds): %s" % (show_next_at_seconds, next_url))
+            upnext.send_next_info(G.args, self._stream_data.playable_item, next_episode, next_url, show_next_at_seconds, self._stream_data.playable_item_parent)
+        except Exception:
+            utils.crunchy_log("_handle_upnext: Cannot send upnext notification", xbmc.LOGERROR)
+
+    def _compute_when_episode_ends(self) -> int:
```

Comment:
Add Pydoc and explain how it works

Reply:
That would make things easier :D Have to look at this more closely, as it is the core of it all and what gave and give me headaches.